internal/services/oracle/oracle_autonomous_database_backup_data_source.go

```diff
@@ -0,0 +1,221 @@
+	IsRestorable                 bool    `tfschema:"is_restorable"`
+	LifecycleDetails             string  `tfschema:"lifecycle_details"`
+	LifecycleState               string  `tfschema:"lifecycle_state"`
+	ProvisioningState            string  `tfschema:"provisioning_state"`
+	TimeAvailableTil             string  `tfschema:"time_available_til"`
+	TimeEnded                    string  `tfschema:"time_ended"`
+	TimeStarted                  string  `tfschema:"time_started"`
+	Location                     string  `tfschema:"location"`
+}
+
+func (a AutonomousDatabaseBackupDataSource) Arguments() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{
+		"autonomous_database_id": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+	}
+}
+
+func (a AutonomousDatabaseBackupDataSource) Attributes() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{
+		"autonomous_database_backups": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"id": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+					"display_name": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"location": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"backup_type": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"retention_period_in_days": {
+						Type:     schema.TypeInt,
+						Computed: true,
+					},
+
+					"autonomous_database_ocid": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"autonomous_database_backup_ocid": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"database_backup_size_in_tbs": {
+						Type:     schema.TypeFloat,
+						Computed: true,
+					},
+
+					"database_version": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"is_automatic": {
+						Type:     schema.TypeBool,
+						Computed: true,
+					},
+
+					"is_restorable": {
+						Type:     schema.TypeBool,
+						Computed: true,
+					},
+
+					"lifecycle_details": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"lifecycle_state": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"provisioning_state": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"time_available_til": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"time_ended": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+
+					"time_started": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (a AutonomousDatabaseBackupDataSource) ModelObject() interface{} {
+	return &AutonomousDatabaseBackupsDataModel{}
+}
+
+func (a AutonomousDatabaseBackupDataSource) ResourceType() string {
+	return "azurerm_oracle_autonomous_database_backup"
```

Comment:
There seems to be a name / functionality mismatch here, the data source implies a single backup, where the implementation returns all the backups on a server. I'd recommend renaming this data source "azurerm_oracle_autonomous_database_backups" and creating another data source for a single backup (like `azurerm_image` and `azurerm_images`)
```suggestion
	return "azurerm_oracle_autonomous_database_backups"
```